packages/cli/src/lookup.ts

```diff
@@ -14,75 +33,130 @@ import {
 
-const sigSpec = new EnumType(["draft-cavage-http-signatures-12", "rfc9421"]);
-
-interface CommandOptions {
-  authorizedFetch?: boolean;
-  firstKnock: "draft-cavage-http-signatures-12" | "rfc9421";
-  traverse?: boolean;
-  suppressErrors?: boolean;
-  raw?: boolean;
-  compact?: boolean;
-  expand?: boolean;
-  userAgent?: string;
-  separator: string;
-  output?: string;
-  timeout?: number;
-}
-
-export async function createFileStream(
-  outputPath: string,
-): Promise<WritableStream> {
-  try {
-    const filepath = isAbsolute(outputPath)
-      ? outputPath
-      : resolve(Deno.env.get("PWD") || Deno.cwd(), outputPath);
-
-    const parentDir = dirname(filepath);
-    await Deno.mkdir(parentDir, { recursive: true });
-
-    const file = await Deno.open(filepath, {
-      write: true,
-      create: true,
-      truncate: true,
-    });
-
-    return new WritableStream({
-      write: (chunk) => file.write(chunk).then(() => {}),
-      close: () => file.close(),
-      abort: (reason) => {
-        file.close();
-        throw reason;
-      },
-    });
-  } catch (err) {
-    const spinner = ora({
-      text: `Failed to write output to ${colors.red(outputPath)}.`,
-      discardStdin: false,
-    });
-    spinner.fail();
-    console.error(`Error: ${String(err)}`);
-
-    if (err instanceof Deno.errors.PermissionDenied) {
-      console.error(
-        "Permission denied. Try running with proper permissions.",
-      );
-    } else if (err instanceof Deno.errors.NotFound) {
-      console.error("Path does not exist or is invalid.");
-    } else if (err instanceof Deno.errors.IsADirectory) {
-      console.error("The specified path is a directory, not a file.");
-    }
-    Deno.exit(1);
+export const authorizedFetchOption = withDefault(
+  object({
+    authorizedFetch: flag("-a", "--authorized-fetch", {
+      description: message`Sign the request with an one-time key.`,
+    }),
+    firstKnock: withDefault(
+      option(
+        "--first-knock",
+        choice(["draft-cavage-http-signatures-12", "rfc9421"]),
+        {
+          description:
+            message`The first-knock spec for -a/--authorized-fetch. It is used for the double-knocking technique.`,
+        },
+      ),
+      "draft-cavage-http-signatures-12" as const,
+    ),
+  }),
+  { authorizedFetch: false } as const,
+);
+
+const traverseOption = withDefault(
+  object({
+    traverse: flag("-t", "--traverse", {
+      description:
+        message`Traverse the given collection to fetch all items. If it is turned on, the argument cannot be multiple.`,
+    }),
+    suppressErrors: option("-S", "--suppress-errors", {
+      description:
+        message`Suppress partial errors while traversing the collection.`,
+    }),
+  }),
+  { traverse: false } as const,
+);
+
+export const lookupCommand = command(
+  "lookup",
+  merge(
+    object({ command: constant("lookup") }),
+    traverseOption,
+    authorizedFetchOption,
+    debugOption,
+    object("lookup", {
```

Comment:
More appropriate labeling (it's shown on the `--help` message):

```suggestion
    object("Looking up options", {
```